php/libraries/Utility.class.inc

```diff
@@ -108,7 +108,8 @@ class Utility
         $factory = NDB_Factory::singleton();
         $DB      = $factory->database();
 
-        $query = "SELECT ConsentID, Name, Label FROM consent";
+        $query = "SELECT ConsentID, Name, Label FROM consent
+                  WHERE ParentID IS NOT NULL";
```

Comment:
Why is the `ParentID IS NOT NULL` constraint added? My understanding from the description is that the parent consent is still a type of consent..

Reply:
in PR #6044, i added a getParentConsents()... a parent consent isn't classified as a consent because no status/date/withdrawal data should be entered in `candidate_consent_rel` table for those consents. they are parents only, not actual consents with consent data

Reply:
ideally `getConsentList()` would be renamed to `getConsentFields()` but then we would have to change every where the function is being used

Reply:
Now I'm really confused about the usage of "Parent Consent" in this PR. Is it a type of consent or is it just a way to group consents?

Reply:
@zaliqarosli 

@driusan might be right here, it seems like the parent consent is not really a consent and thus should not live in the same table. it seems like what is being added is a consent category but that cant live in the same table.

To be in the consent table it needs to have a similar yes/no option to all it's children. that could be done technically where if you answer yes/no to the parent it would apply to all children as well... further design might be needed to ensure that if you later add a consent field to a parent it should get autofilled if the parent is set to yes... details which could be figured out later

Reply:
I based this method of parenting/grouping on how the parent/child relationship works for ConfigSettings ....

Reply:
`ConfigSettings` is done that way because of legacy from when all the settings were in `config.xml` and ported to the database. Since it's was an XML document, both the parent and child *are* valid config settings (getSetting() on the parent being an array of child values.). If it was designed from scratch it wouldn't be done that way..

Reply:
hmm okay so are we thinking a new parent_consent table?

Reply:
no, a ConsentGroup table.

Reply:
i don't think i understand the difference. what makes it a "real parent"?

Reply:
It's really a parent if both the parent and child are types of consent and they form a tree. It's not really a parent if only the children are types of consent and the parent is just to group them.

Reply:
righhttt. so for ConfigSettings, are the parent configs real parent configs or no?

Reply:
in some ways, the parent consents are consents themselves i.e. they are the whole ICF, instead of individual consent fields

Reply:
if you define a consent as having a date of "signature" a potential date of withdrawal and a yes/no status, then in the current code the parent is not a "real parent", the parent is simply not a consent it's just a group.